Doppler.Currency/Services/CurrencyService.cs

```diff
@@ -1,22 +1,22 @@
 ﻿using System;
-using System.Collections.Generic;
 using System.Threading.Tasks;
 using CrossCutting;
 using Doppler.Currency.Dtos;
 using Doppler.Currency.Enums;
+using Doppler.Currency.Factory;
 using Microsoft.Extensions.Logging;
 
 namespace Doppler.Currency.Services
 {
     public class CurrencyService : ICurrencyService
     {
         private readonly ILogger<CurrencyService> _logger;
-        private readonly IReadOnlyDictionary<CurrencyCodeEnum, CurrencyHandler> _currencyHandlers;
+        private readonly ICurrencyFactory _currencyFactory;
 
         public CurrencyService(
             ILogger<CurrencyService> logger,
-            IReadOnlyDictionary<CurrencyCodeEnum, CurrencyHandler> currencyHandlers) =>
-            (_logger, _currencyHandlers) = (logger, currencyHandlers);
+            ICurrencyFactory currencyFactory) =>
+            (_logger, _currencyFactory) = (logger, currencyFactory);
```

Comment:
😮 Oh, I never used this way to write constructors that only set properties, interesting...